pool/test/message_history.ml

```diff
@@ -0,0 +1,289 @@
+  in
+  Lwt.return_unit
+;;
+
+let find_experiment () =
+  Experiment.find database_label experiment_id ||> get_exn
+;;
+
+let find_contact () = Contact.find database_label contact_id ||> get_exn
+let find_session () = Session.find database_label session_id ||> get_exn
+
+let find_assignment () =
+  Assignment.find database_label assignment_id ||> get_exn
+;;
+
+let sort_entity_uuids = CCList.sort Pool_common.Id.compare
+
+let sort_history history =
+  History.
+    { history with
+      entity_uuids = CCList.sort Pool_common.Id.compare history.entity_uuids
+    }
+;;
+
+let check_history_create =
+  Alcotest.(check (option Test_utils.message_history_crate) "succeeds")
+;;
+
+let with_sorted_entity_uuids =
+  let open CCFun.Infix in
+  Email.job_message_history %> CCOption.map sort_history
+;;
+
+let txt_msg_with_sorted_entity_uuids =
+  let open CCFun.Infix in
+  Text_message.job_message_history %> CCOption.map sort_history
+;;
+
+let create_history label entity_uuids =
+  Some
+    History.
+      { message_template = Some Label.(show label)
+      ; entity_uuids = sort_entity_uuids entity_uuids
+      }
+;;
+
+let account_suspension_notification _ () =
+  let%lwt contact = find_contact () in
+  let%lwt res =
+    AccountSuspensionNotification.create tenant (Contact.user contact)
+    ||> get_exn
+    ||> with_sorted_entity_uuids
+  in
+  let expected =
+    [ Contact.Id.to_common contact_id ]
+    |> create_history Label.AccountSuspensionNotification
+  in
+  let () = check_history_create expected res in
+  Lwt.return_unit
+;;
+
+let assignment_confirmation _ () =
+  let%lwt contact = find_contact () in
+  let%lwt experiment = find_experiment () in
+  let%lwt session = find_session () in
+  let%lwt assignment = find_assignment () in
+  let%lwt res =
+    AssignmentConfirmation.prepare tenant contact experiment session None
+    ||> fun create -> create assignment |> with_sorted_entity_uuids
+  in
+  let expected =
+    [ Experiment.Id.to_common experiment_id
+    ; Session.Id.to_common session_id
+    ; Contact.Id.to_common contact_id
+    ]
+    |> create_history Label.AssignmentConfirmation
+  in
+  let () = check_history_create expected res in
+  Lwt.return_unit
+;;
+
+let assignment_session_change _ () =
+  let%lwt contact = find_contact () in
+  let%lwt experiment = find_experiment () in
+  let%lwt session = find_session () in
+  let%lwt new_session = SessionRepo.create experiment_id () in
+  let%lwt assignment = find_assignment () in
+  let message =
+    Test_utils.Model.create_manual_message
+      ~recipient:(Contact.email_address contact)
+      ()
+  in
+  let%lwt res =
+    AssignmentSessionChange.create
+      message
+      tenant
+      experiment
+      ~new_session
+      ~old_session:session
+      assignment
+    ||> with_sorted_entity_uuids
+  in
+  let expected =
+    [ Experiment.Id.to_common experiment_id
+    ; Session.Id.to_common session_id
+    ; Session.(Id.to_common new_session.id)
```

Comment:
```suggestion
    ; Session.Id.to_common new_session.id
```

Reply:
This will trigger a warning `Error (warning 40): id was selected from type Session.t.`.

The alternative would be `Session.Id.to_common new_session.Session.id`